cmd/simpleargs/main.go

```diff
@@ -0,0 +1,141 @@
+var output string
+var owner string
+var repo string
+var prnumber int
+var releaseVersion string
+
+func writeArgs(arglist []string) {
+	for i, arg := range arglist[1:] {
+		data := []byte(arg)
+		filename := fmt.Sprintf("ARG_%d", i)
+		err := ioutil.WriteFile(filepath.Join(output, filename), data, 0644)
+		log.Printf(filepath.Join(output, filename))
+		if err != nil {
+			log.Fatalln(err)
+		}
+	}
+}
+
+func memberValidation(authorAssociation string) error {
+
+	if (authorAssociation != "COLLABORATOR") && (authorAssociation != "MEMBER") {
+		return fmt.Errorf("Not a member or collaborator")
+	} 
+	return nil
+}
+
+func regexValidation(regex string,comment string) ([]string, error) {
+	argRe := regexp.MustCompile(regex)
+	if argRe.MatchString(comment) {
+		arglist := argRe.FindStringSubmatch(comment)
+		return arglist, nil
+	} else {
+		return make([]string,0),fmt.Errorf("Invalid command")
+	}
+}
+
+func postComment(client *github.Client, comment string) error {
+	issueComment := &github.IssueComment{Body: github.String(comment)}
+	issueComment, _, err := client.Issues.CreateComment(context.Background(), owner, repo, prnumber, issueComment)
+	return err
+}
+
+
+func main() {
+	app := kingpin.New(filepath.Base(os.Args[0]), "simpleargs github comment extract")
+	app.Flag("input", "path to event.json").Default("/github/workflow/event.json").StringVar(&input)
+	app.Flag("output", "path to write args to").Default("/github/home").StringVar(&output)
+	app.Arg("regex", "Regex pattern to match").Required().StringVar(&regex)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	//Github client for posting comments
+	token := os.Getenv("GITHUB_TOKEN")
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+
+	//Reading event.json
+	os.MkdirAll(output, os.ModePerm)
+	data, err := ioutil.ReadFile(input)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	//Parsing event.json
+	event, err := github.ParseWebHook("issue_comment", data)
+	if err != nil {
+		log.Fatalln("could not parse = %v\n", err)
+	}
+	
+	//Checking author association and saving args to file
+	switch e := event.(type) {
+	case *github.IssueCommentEvent:
+		err := memberValidation(*e.GetComment().AuthorAssociation) //check author association
+		if err != nil {
+			log.Printf("Author is not a member or collaborator")
+			os.Exit(78)
+		} 
+		log.Printf("Author is member or collaborator")
+		
+		arglist, err := regexValidation(regex, *e.GetComment().Body) //Validate comment 
+		if err != nil{
+			log.Printf("Matching command not found")
+			os.Exit(78)
+		}
+
+		owner = *e.GetRepo().Owner.Login  //Get parameters
+		repo = *e.GetRepo().Name
+		prnumber = *e.GetIssue().Number
+		releaseVersion = arglist[1]
+		
+		arglist = append(arglist, strconv.Itoa(prnumber))
+		writeArgs(arglist) //Save args to file(releaseVersion in ARG_0, prnumber in ARG_1)
+
+		//Posting benchmark start comment
```

Comment:
Please read more about golang idioms for comments.
Capital letter and full stop for all comments and run go fmt to format the code correctly.

Best to install and use vscode as it does this for you with every save.